spyder/plugins/projects/projecttypes/__init__.py

```diff
@@ -75,11 +78,6 @@ def get_recent_files(self):
         recent_files = [recent_file if os.path.isabs(recent_file)
                         else os.path.join(self.root_path, recent_file)
                         for recent_file in recent_files]
-
-        for recent_file in recent_files[:]:
-            if not os.path.isfile(recent_file):
-                recent_files.remove(recent_file)
-
```

Comment:
Why did you remove these lines? Without these lines `get_recent_files` does nothing.

Reply:
I made this change (removing the validation of the path being a file) to check the tests (`test_set_get_project_filenames_when_closing` and `test_set_get_project_filenames_when_switching`). Seems like the existing tests were passing because of the mutability property of a list (the `set_recent_files` method was changing directly the `opened_files` list that at the end is the object passed by parameter, so the [assert](https://github.com/spyder-ide/spyder/blob/aafb1c21f6e8c58997e43ccfc8cf8d06e2feb9c6/spyder/plugins/projects/tests/test_plugin.py#L197) was checking the return with a list changed by the `set_recent_files` method (the recent files in the tests always were an empty list, since the paths we are testing here aren't actually created).

However, probably we need here to change the tests to create the files and add again the check for the path being a file. So this is not ready yet :)

Reply:
> So this is not ready yet :)

@dalthviz, then please don't forget to add WIP to the PR title.

Reply:
Yep :+1: